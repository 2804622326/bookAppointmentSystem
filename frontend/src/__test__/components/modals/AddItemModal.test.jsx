import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import AddItemModal from '../../../components/modals/AddItemModal.jsx';

test('adds item and closes modal', () => {
  const save = jest.fn();
  const close = jest.fn();
  const { asFragment } = render(
    <AddItemModal show handleClose={close} handleSave={save} itemLabel="Category" />
  );
  fireEvent.change(screen.getByPlaceholderText(/enter category name/i), {
    target: { value: 'Cat' },
  });
  fireEvent.click(screen.getByText('Add'));
  expect(save).toHaveBeenCalledWith('Cat');
  expect(close).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
