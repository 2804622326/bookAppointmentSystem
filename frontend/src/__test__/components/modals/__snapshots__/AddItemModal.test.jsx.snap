// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds item and closes modal 1`] = `<DocumentFragment />`;
